api/routes.go

```diff
@@ -2,60 +2,48 @@ package api
 
 import (
 	"encoding/json"
-	"fmt"
 	"math"
 	"net/http"
 	"sort"
 	"strconv"
 	"time"
-
 	// MySQL driver
-	"gopkg.in/cas.v1"
-
 	log "github.com/Sirupsen/logrus"
 	"github.com/gorilla/mux"
-
 	"github.com/wtg/shuttletracker/model"
+	"gopkg.in/cas.v1"
 )
 
 //RouteIsActive determines if the current time means a route should be active or not
 //TODO: Move this to updater
 func (api *API) RouteIsActive(r *model.Route) bool {
 
 	//This is a time offset, to ensure routes are activated on the minute they are assigned activate
-	currentTime := time.Now()
-	day := currentTime.Weekday()
+	var currentTime model.Time
+	currentTime.FromTime(time.Now())
+	currentTime.Day = time.Now().Weekday()
 	state := -1
 
-	for idx := range r.TimeInterval {
-		//Bring the date to the current day to compare only the time object
-		add := -(int(r.TimeInterval[idx].Time.Sub(time.Now().Truncate(24*time.Hour)).Hours()/24) - 1)
-		r.TimeInterval[idx].Time = r.TimeInterval[idx].Time.AddDate(0, 0, add)
+	if r.TimeInterval == nil || len(r.TimeInterval) == 1 {
+		state = 1
 	}
 	for idx, val := range r.TimeInterval {
+		//If it is the last in the time list (latest time for the week) use this index
 		if idx >= len(r.TimeInterval)-1 {
 			state = val.State
 			break
 		} else {
-			if day < val.Day {
+			if currentTime.After(val) && currentTime.After(r.TimeInterval[idx+1]) {
 				continue
-			} else if day == val.Day && !currentTime.After(val.Time) {
-				continue
-			} else if day == val.Day && currentTime.After(r.TimeInterval[idx+1].Time) {
-				continue
-			} else if day > val.Day && day <= r.TimeInterval[idx+1].Day {
-				if currentTime.After(r.TimeInterval[idx+1].Time) {
-					continue
-				}
 			}
 			state = val.State
 			break
-
 		}
 	}
 
 	route := model.Route{}
-	if(api.db != nil){
+	//Check if db is nil for testing
+	if api.db != nil {
```

Comment:
We should construct an instance of `API` with a mock database. The code shouldn't be branching differently to handle being tested.